### PR TITLE
Field Picker Story Revamp

### PR DESCRIPTION
**Pull request by ElliotTheHuman**
### :sparkles: Changes

- `AccordionDisclosure` added "SimpleLayoutProps"
- Updated `FieldPicker.story` to match Explore team's latest specs
  - https://www.figma.com/file/o8vnoAdtk1fbf99QJDxTFG/Explore-Field-List?node-id=6261%3A133

### :white_check_mark: Requirements

- [ ] Includes test coverage for all changes
- [ ] Build and tests are passing
- [ ] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [x] PR is ideally < ~400LOC

### :camera: Screenshots
image (3)

### CODE REVIEW

File: packages/components/src/Tree/stories/FieldPicker.story.tsx [hunk 1]
```diff
@@ -112,18 +115,15 @@ const PickerItem = ({ children = 'Cost', truncate = false }) => {
...
-    detailAccessory
-    indicatorIcons={{ close: 'CaretRight', open: 'CaretDown' }}
-    indicatorPosition="right"
-    label={children}
-  >
+// Wrapper for Tree and TreeItem elements that doesn't render an actual Tree
+const FauxTree: FC<{ children: ReactNode }> = ({ children }) => (
+  <TreeStyle depth={-1} hovered={false}>
+    <TreeContext.Provider value={{ depth: 0 }}>{children}</TreeContext.Provider>
+  </TreeStyle>
```
Comment (lukelooker) [inline packages/components/src/Tree/stories/FieldPicker.story.tsx:122]: As discussed make `hoverered` an optional prop

Create this composition as a pattern we export for this specific use case.

With that done would it be possible to stop exporting `TreeStyle` and `TextContext` from the package?

File: packages/components/src/Tree/stories/FieldPicker.story.tsx [hunk 2]
```diff
@@ -135,20 +135,40 @@ const ViewTree: FC<{ children: string; defaultOpen?: boolean }> = ({
...
+    indicatorSize="xxsmall"
+    indicatorIcons={{ close: 'CaretRight', open: 'CaretDown' }}
+  >
+    <AccordionDisclosure px="4px" py="0" height="24px" color="text4">
+      <FlexItem flex="1" fontSize="xsmall" lineHeight="xsmall">
+        <Space between>
+          <Truncate>{label}</Truncate>
+          <Badge intent="inform">1</Badge>
+        </Space>
+      </FlexItem>
```
Comment (lukelooker) [inline packages/components/src/Tree/stories/FieldPicker.story.tsx:160]:
```suggestion
      <Space between>
        <Text fontSize="xsmall"><Truncate>{label}</Truncate></Text>
        <Badge intent="inform">1</Badge>
      </Space>
```

**Review by lukelooker — CHANGES_REQUESTED** [review]

File: packages/components/src/Tree/stories/FieldPicker.story.tsx (hunk 1 above)
Comment (ElliotTheHuman) [reply packages/components/src/Tree/stories/FieldPicker.story.tsx]: Yep! Went ahead and removed `TreeContext` from the package exports and replaced it with a `TreeIndentManager` export instead (`TreeStyle` was not being exported originally).

File: packages/components/src/Tree/stories/FieldPicker.story.tsx [hunk 3]
```diff
@@ -135,20 +126,40 @@ const ViewTree: FC<{ children: string; defaultOpen?: boolean }> = ({
...
+    indicatorIcons={{ close: 'CaretRight', open: 'CaretDown' }}
+  >
+    <AccordionDisclosure px="4px" py="0" height="24px" color="text4">
+      <Space between>
+        <Text fontSize="xsmall">
+          <Truncate>{label}</Truncate>
+        </Text>
+        <Badge intent="inform">1</Badge>
+      </Space>
+    </AccordionDisclosure>
```
Comment (lukelooker) [inline packages/components/src/Tree/stories/FieldPicker.story.tsx:152]:
```suggestion
    <AccordionDisclosure px="xxsmall" py="none" fontSize="xsmall">
      <Space between>
        <Truncate>{label}</Truncate>
        <Badge intent="inform">1</Badge>
      </Space>
    </AccordionDisclosure>
```

**Review by lukelooker — COMMENTED** [review]
Haven't had a chance to fully review but had one suggestion (see inline comment)

**Review by lukelooker — APPROVED** [review]
tree

**Pull request merged**